repair/row_level.cc

```diff
@@ -846,13 +846,19 @@ class repair_meta {
 
 public:
     future<> stop() {
+        // Prevent double-stop
+        if (!_repair_writer) {
+            assert(_gate.is_closed());
+            return make_ready_future<>();
+        }
```

Comment:
double-stop is illegal, how can it happen?

Reply:
Once from `repair_row_level_stop` and the other from `auto_stop_master` (that used to only call `close()` but I changed it in this patch to `stop()` instead)

Reply:
I think it's also wrong: the first stop() starts the stopping process, the second stop() returns a ready future before the first stop() completes.

Reply:
it's a good point.  will fix.

Reply:
Fixed too in v5 (force-pushed)